db/Controllers/api.controller.js

```diff
@@ -1,38 +1,50 @@
-const endpoints = require('../../endpoints.json')
-const {readTopics, readArticleId} = require('../models/api.model')
+const endpoints = require("../../endpoints.json");
+const {
+  readTopics,
+  readArticleId,
+  readArticles,
+} = require("../models/api.model");
 
 exports.getApiDocu = (req, res) => {
-    res.status(200).send({endpoints})
-}
+  res.status(200).send({ endpoints });
+};
 
 exports.getTopics = (req, res) => {
-        readTopics().then((topics) => {
-            res.status(200).send({topics})
-        })
-        .catch((err) => {
-            console.error("Error fetching topics:", err);
-            res.status(500).send({msg: "server error"})
-        })
-}
+  readTopics()
+    .then((topics) => {
+      res.status(200).send({ topics });
+    })
+    .catch((err) => {
+      console.error("Error fetching topics:", err);
+      res.status(500).send({ msg: "server error" });
+    });
+};
 
 exports.getArticleId = (req, res) => {
+  const { article_id } = req.params;
 
-    const { article_id } = req.params;
-
-    readArticleId(article_id)
+  readArticleId(article_id)
     .then((article) => {
-        if (!article) {
-            return res.status(404).send({msg: "Article not found"})
-        } else{
-            res.status(200).send({ article })
-        }
-
+      if (!article) {
+        return res.status(404).send({ msg: "Article not found" });
+      } else {
+        res.status(200).send({ article });
+      }
     })
     .catch((err) => {
-        if(err.code === '22P02') {
-            res.status(400).send({msg: "Invalid article ID"})
-        }else{
-            res.status(500).send({msg: "Internal server error"})
-        }
-    })
-}
\ No newline at end of file
+      if (err.code === "22P02") {
+        res.status(400).send({ msg: "Invalid article ID" });
+      } else {
+        res.status(500).send({ msg: "Internal server error" });
+      }
+    });
```

Comment:
Instead of dealing with these errors here you should extract them. Use next(err) and send the error to an error handler in app.js. 

The big advantage is that otherwise you will have to repeat this code for lots of controllers. If in every controller you catch the error and send it to next - you can have one place that deals with the same code as above which will definitely come up again 😄